sys/hash_string/hash_string.c

```diff
@@ -26,7 +26,7 @@ unsigned long hash_string(unsigned char *str)
     unsigned long hash = 5381;
     int c;
 
-    while ((c = *str++)) {
+    for (char c; (c = *str) != 0; ++str) {
```

Comment:
But in general: this change is a cosmetic change that is not really in the scope of this PR, so I would prefer to not add any changes to this file at all.